p2p/simulations/adapters/inproc.go

```diff
@@ -364,9 +369,27 @@ func socketPipe() (net.Conn, net.Conn, error) {
 	if err != nil {
 		return nil, nil, err
 	}
+
+	setSocketBuffer(pipe1)
+	setSocketBuffer(pipe2)
+
 	return pipe1, pipe2, nil
 }
 
+func setSocketBuffer(conn net.Conn) {
+	switch v := conn.(type) {
+	case *net.UnixConn:
+		err := v.SetReadBuffer(socketReadBuffer)
+		if err != nil {
+			panic(err)
```

Comment:
We should either return an error or ignore it, a panic seems overkill.

Reply:
You're right, fixed it.

Even though this is probably not a critical error, I decided to return it, rather than ignore it, so that if someone tries to update the buffer with a size that is too big, they immediately see it fail.